Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs

```diff
@@ -862,12 +867,29 @@ data Action = NoAction
             | RepeatRideCountDown Int String String String
             | StopRepeatRideTimer 
             | OpenLiveDashboard
-            | UpdatePeekHeight
+            | UpdatePeekHeight 
+            | AutoScrollCountDown Int String String String
+            | StopAutoScrollTimer 
+            | UpdateRepeatTrips RideListWithStatusRes String
+            | RemoveShimmer 
 
 eval :: Action -> HomeScreenState -> Eval Action ScreenOutput HomeScreenState
 
 eval ShowMoreSuggestions state = continue state { props {suggestionsListExpanded = not state.props.suggestionsListExpanded} }
 
+eval RemoveShimmer state = continue state{props{showShimmer = false}}
+
+eval (UpdateRepeatTrips rideList status) state = do
+    let shimmerState = state{props{showShimmer = false}}
+    case status of
+        "success" -> do
+            let list = rideListToTripsTransformer (rideList ^._list)
+            unless (null list) $ do
+                let updatedMap = updateMapWithPastTrips list shimmerState
```

Comment:
`unless` won't work in the `eval`. can you check once. It will be always executed.